### ISSUE TITLE
how to deal with the "mask  data" in SHAP

### ISSUE THREAD

**Issue by koyurion** [shap/shap#1870]
Hi, When I try to use **SHAP's GradientExplainer** to explain my deeplearning model in **Pytorch** framework, I am not sure how to deal with the mask data which are needed to pass in the forward call and i only want to obtain **the shap value of xTrainFeature_part1**. As follow,
`
class myModel(nn.Module):

    def __init__(self,...):
        super(myModel, self).__init__()
        ....

    def forward(self, xTrainFeature_part1,xTrainFeature_part2,mask):
        ...
        # some graph convolution operation to update features
        ...
        # make the padded features to zero
        fea = fea * mask
        ...

        return y_pred
        
`
But, when I check the source code of GradientExplainer, I find that it wolud **randomize the test data and background data**:

![未命名图片](https://user-images.githubusercontent.com/32454333/110631022-0e271880-81e1-11eb-9228-74d1de00e1d5.png)

So, can I pass the mask data to the forward function by "torch.save()" and "torch.load()" ?

**Comment by koyurion** [comment]
你好，你的邮件已收到。谢谢！
